app/models/approvals/parallel.rb

```diff
@@ -0,0 +1,56 @@
+# A node in an approval chain that allows its child approvals to come in in
+# any order
+module Approvals
+  class Parallel < Approval
+    validates :min_children_needed, numericality: {allow_blank: true}
+
+    workflow do
+      on_transition { self.touch } # https://github.com/geekq/workflow/issues/96
+
+      state :pending do
+        event :initialize, transitions_to: :actionable
+      end
+
+      state :actionable do
+        # on_actionable_entry, below
+
+        event :initialize, transitions_to: :actionable do
+          halt  # prevent state transition
+        end
+        event :child_approved, transitions_to: :approved do |_|
+          halt unless self.children_approved?
+        end
+        event :force_approve, transitions_to: :approved
+      end
+
+      state :approved do
+        on_entry { self.notify_parent_approved }
+
+        event :initialize, transitions_to: :approved do
+          self.notify_parent_approved
+          halt  # prevent state transition
+        end
+
+        event :child_approved, transitions_to: :approved do |_|
+          halt  # additional approvals do nothing
+        end
+      end
+    end
+
+    def on_actionable_entry(old_state, event)
+      if self.child_approvals.exists?
+        self.child_approvals.each(&:initialize!)
+      else
+        self.force_approve!
+      end
```

Comment:
I don't think that restriction gains us anything. Why not tackle the general case?